pkg/console/subresource/consoleserver/types.go

```diff
@@ -100,10 +100,27 @@ type ProjectAccess struct {
 	AvailableClusterRoles []string `yaml:"availableClusterRoles,omitempty"`
 }
 
+// If sub-catalog type(s) is(are) mentioned in both `enabled` and `disabled` list, the sub-catalog type(s) should be hidden.
+type DeveloperConsoleCatalogTypes struct {
+	// enabled is a list of developer catalog types (sub-catalogs IDs) that will be shown to users.
+	// Types (sub-catalogs) are added via console plugins, the available types (sub-catalog IDs) are available
+	// in the console on the cluster configuration page, or when editing the YAML in the console.
+	// A new type will not be shown to the user until it is added in the enabled list.
+	// If `enabled` list is empty, complete developer catalog should be hidden.
+	Enabled []string `yaml:"enabled"`
+	// disabled is a list of developer catalog types (sub-catalogs IDs) that are not shown to users.
+	// Types (sub-catalogs) are added via console plugins, the available types (sub-catalog IDs) are available
+	// in the console on the cluster configuration page, or when editing the YAML in the console.
+	// If all the available sub-catalog types are added in the `disabled` list, complete developer catalog should be hidden.
+	Disabled []string `yaml:"disabled,omitempty"`
```

Comment:
why we need list of disabled types ? I would assume that if the type is not enabled, then its disabled?

Reply:
We considered both because, for ex, if there are 10 sub-catalog types and the admin wants to enable 8 of them, then instead of listing all the types in `enabled` the admin can simply list the remaining 2 in the disabled list.